lib/src/controls/yaru_popup_menu_button.dart

```diff
@@ -102,3 +102,163 @@ class _YaruPopupDecoration extends StatelessWidget {
+    required this.child,
+    required this.onTap,
+    this.enabled = true,
+  });
+
+  final Set<T> values;
+  final void Function() onTap;
+  final T value;
+  final Widget child;
+  final EdgeInsets contentPadding;
+  final bool enabled;
+
+  @override
+  State<YaruMultiSelectItem> createState() => _YaruMultiSelectItemState();
+}
+
+class _YaruMultiSelectItemState extends State<YaruMultiSelectItem> {
+  @override
+  Widget build(BuildContext context) {
+    final textColor = widget.enabled
+        ? Theme.of(context).colorScheme.onSurface
+        : Theme.of(context).disabledColor;
+    final borderColor = Theme.of(context).colorScheme.onSurface.withOpacity(
+          Theme.of(context).brightness == Brightness.light ? 0.4 : 1,
+        );
+    return ListTile(
+      visualDensity: VisualDensity.compact,
+      enabled: widget.enabled,
+      contentPadding: widget.contentPadding,
+      onTap: () {
+        setState(() {
+          widget.onTap();
+        });
+      },
+      leading: widget.values.contains(widget.value)
+          ? AnimatedContainer(
+              height: 18,
+              width: 18,
+              duration: const Duration(
+                milliseconds: 200,
+              ),
+              decoration: BoxDecoration(
+                color: Theme.of(context).primaryColor,
+                borderRadius: BorderRadius.circular(4),
+              ),
+              child: const Icon(
+                Icons.check,
+                color: Colors.white,
+                size: 18,
+              ),
+            )
+          : AnimatedContainer(
+              height: 18,
+              width: 18,
+              duration: const Duration(
+                milliseconds: 200,
+              ),
+              decoration: BoxDecoration(
+                border: Border.all(
+                  color: borderColor,
+                  width: 2,
+                ),
+                borderRadius: BorderRadius.circular(4),
+              ),
+            ),
+      title: DefaultTextStyle(
+        style: TextStyle(color: textColor, fontSize: 16),
+        child: widget.child,
+      ),
+    );
+  }
+}
+
+class YaruCheckMenuItem<T> extends PopupMenuItem<T> {
+  /// Creates a popup menu item with a checkmark.
+  ///
+  /// By default, the menu item is [enabled] but unchecked. To mark the item as
+  /// checked, set [checked] to true.
+  ///
+  /// The `checked` and `enabled` arguments must not be null.
+  const YaruCheckMenuItem({
+    super.key,
+    super.value,
+    this.checked = false,
+    super.enabled,
+    super.padding,
+    super.height,
+    super.mouseCursor,
+    super.child,
+  });
+
+  final bool checked;
+
+  @override
+  Widget? get child => super.child;
+
+  @override
+  PopupMenuItemState<T, YaruCheckMenuItem<T>> createState() =>
+      _CheckedPopupMenuItemState<T>();
+}
+
+class _CheckedPopupMenuItemState<T>
+    extends PopupMenuItemState<T, YaruCheckMenuItem<T>>
+    with SingleTickerProviderStateMixin {
```

Comment:
`SingleTickerProviderStateMixin` can be removed if we're not dealing with raw `AnimationControllers`